src/service/xhr-impl.js

```diff
@@ -576,6 +593,13 @@ export class FetchResponseHeaders {
  * @return {!Xhr}
  */
 export function xhrServiceForTesting(window) {
+  registerServiceBuilder(window, 'ampdoc', function() {
+    return {
+      isSingleDoc() {
+        return false;
+      },
+    };
+  });
```

Comment:
I'd prefer to resolve this. I'm ok with not calling this code in the test code as well. Otherwise this will be asking for a trouble.

Ditto: otherwise LGTM.